src/domain/accounts/counterfactual-safes/entities/counterfactual-safe.entity.ts

```diff
@@ -0,0 +1,18 @@
+import { RowSchema } from '@/datasources/db/entities/row.entity';
+import { AccountSchema } from '@/domain/accounts/entities/account.entity';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
+
+export type CounterfactualSafe = z.infer<typeof CounterfactualSafeSchema>;
+
+export const CounterfactualSafeSchema = RowSchema.extend({
+  chain_id: z.string(),
+  creator: AddressSchema,
+  fallback_handler: AddressSchema,
+  owners: z.array(AddressSchema),
+  predicted_address: AddressSchema,
+  salt_nonce: z.string(),
+  singleton_address: AddressSchema,
+  threshold: z.number().int().gte(0),
```

Comment:
```suggestion
  threshold: z.number().int().gte(1),
```

Reply:
Consider the minimum value, maybe we should also enforce a minimum number of `owner` elements as well. What do you think?

Reply:
Good point, done in [f2ca4c0](https://github.com/safe-global/safe-client-gateway/pull/1773/commits/f2ca4c0847951435d54b09bde99a34f1bf63c0ea)